packages/vue/src/components/search/DataSearch.jsx

```diff
@@ -262,15 +264,31 @@ const DataSearch = {
 		},
 	},
 	methods: {
+		validateDataField() {
+			const propName = 'dataField';
+			const componentName = 'DataSearch';
+			const props = this.$props;
+			const requiredError = `${propName} supplied to ${componentName} is required. Validation failed.`;
+			const propValue = props[propName];
+			if (this.config && !this.config.enableAppbase) {
+				if (!propValue) {
+					console.error(requiredError);
+					return;
+				}
+				if (typeof propValue !== 'string' && !Array.isArray(propValue)) {
+					console.error(
+						`Invalid ${propName} supplied to ${componentName}. Validation failed.`,
+					);
+					return;
+				}
+				if (Array.isArray(propValue) && propValue.length === 0) {
+					console.error(requiredError);
+				}
+			}
+		},
 		updateQueryOptions() {
-			if (
-				this.customHighlight
-				&& typeof this.customHighlight === 'function'
-			) {
-				this.setCustomHighlightOptions(
-					this.componentId,
-					this.customHighlight(this.$props),
-				);
+			if (this.customHighlight && typeof this.customHighlight === 'function') {
```

Comment:
@ShahAnuj2610 Why formatting changes?

Reply:
not sure. I'm using prettier and eslint as per repo config.